### PR TITLE
Added Missing APIS

### PR DESCRIPTION
**Pull request by RangaKoushik829**
To implement missing User Profile APIs in NPM package

### CODE REVIEW

**Comment by darshanclevertap** [comment]
@rangakoushik As discussed, please add the comments for all new public APIs, methods added in this PR

File: package.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
   },
   "homepage": "https://github.com/CleverTap/clevertap-node",
   "dependencies": {
+    "clevertap": "^1.2.0",
```
Comment (korolev) [inline package.json:28]: Why does this dependency appear?

File: lib/clevertap.js [hunk 2]
```diff
@@ -373,6 +374,423 @@ CleverTap.prototype.events = function(query, options, callback) {
...
+        if (error) {
+            console.error(error);
+            if (callback && typeof callback === 'function') {
+                callback(null);
+            }
+            reject(error);
+            return;
+        }
+
+        this.api.profile(query, options, (res) => {
```
Comment (korolev) [inline lib/clevertap.js]: looks like `this.api.delete` should be here. And we have to send the correct endpoint as a first argument

**Comment by KambleSonam** [comment]
Closing this PR and raising new PR with all fixes

**Pull request closed without merging**